src/javascript/app_2/App/Components/Elements/Tooltip2/tooltip-bubble.jsx

```diff
@@ -0,0 +1,112 @@
+
+const FadeIn = posed.div({
+    enter: {
+        opacity: 1,
+
+        transition: {
+            duration: 300,
+        },
+    },
+    exit: {
+        opacity: 0,
+
+        transition: {
+            duration: 300,
+        },
+    },
+});
+
+class TooltipBubble extends React.PureComponent {
+
+    getHorizontalCenter = rectangle => rectangle.left + (rectangle.width / 2)
+
+    getVerticalCenter = rectangle => rectangle.top + (rectangle.height / 2)
+
+    getBubblePositionStyle = (alignment, tooltip_trigger_rectangle) => {
+        switch (alignment) {
+            case 'top': return {
+                left     : this.getHorizontalCenter(tooltip_trigger_rectangle),
+                transform: 'translateX(-50%)',
+                bottom   : `calc(100% - ${tooltip_trigger_rectangle.top}px)`,
+            };
+            case 'right': return {
+                left     : tooltip_trigger_rectangle.x + tooltip_trigger_rectangle.width,
+                top      : this.getVerticalCenter(tooltip_trigger_rectangle),
+                transform: 'translateY(-50%)',
+            };
+            case 'bottom': return {
+                left     : this.getHorizontalCenter(tooltip_trigger_rectangle),
+                transform: 'translateX(-50%)',
+                top      : tooltip_trigger_rectangle.y + tooltip_trigger_rectangle.height,
+            };
+            case 'left': return {
+                right    : `calc(100% - ${tooltip_trigger_rectangle.left}px)`,
+                top      : this.getVerticalCenter(tooltip_trigger_rectangle),
+                transform: 'translateY(-50%)',
+            };
+            default: return {
+                left: tooltip_trigger_rectangle.x,
+                top : tooltip_trigger_rectangle.y,
+            };
+        }
+    }
+
+    render() {
+        const {
+            alignment,
+            icon,
+            message,
+            tooltip_trigger_rectangle,
+        } = this.props;
+    
+        return ReactDOM.createPortal(
+            <PoseGroup>
+                <FadeIn key='fade_in' initialPose='exit'>
+                    <span
+                        style={this.getBubblePositionStyle(
+                            alignment,
+                            tooltip_trigger_rectangle,
+                        )}
+                        className={classNames(
+                            'tooltip-2__bubble',
+                            `tooltip-2__bubble--${alignment}`,
+                        )}
+                    >
+                        <span className={classNames(
+                            'tooltip-2__bubble__arrow',
+                            `tooltip-2__bubble__arrow--${alignment}`,
+                        )}
+                        />
+                        { icon &&
+                            <span className='tooltip-2__bubble__icon'>
+                                {icon === 'info'     && <Icon icon={IconInfoBlue} />}
```

Comment:
Please add `parentheses`, for readability's sake. Parentheses in conditionals, even if they are not strictly required help define what is clear and correct.